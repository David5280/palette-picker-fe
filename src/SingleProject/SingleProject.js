import React, { Component } from 'react'
import ColorPreview from '../ColorPreview/ColorPreview';
import { serverCall } from '../fetchCalls/fetchCalls';
import { Link } from 'react-browser-router';

export class SingleProject extends Component {
  constructor() {
    super()
    this.state ={
      project_name:'',
      palettes: []
    }
  }

  async componentDidMount() {
    const projectInfo = await serverCall(`projects/${this.props.id}`)
    this.setState({project_id: projectInfo.id, palettes: projectInfo.palette, project_name: projectInfo.name})
  }

  patchProject = async (path, value) => {
    console.log(path)
    await serverCall(path, "PATCH", value)
  }

  deletePalette = async (paletteID) => {
    const deleteRes = await serverCall(`palettes/${paletteID}`, 'DELETE')
    console.log(await deleteRes)
    if (await deleteRes.deleted) {
      const filteredPalettes = this.state.palettes.filter(palette => {
        return palette.id !== paletteID
      })
      this.setState({ palettes: filteredPalettes })
    }
  } 

  showPalettes = () => {
    return this.state.palettes.map((palette, i) => {
      let colors = [
        {
          paletteID: palette.id,
          paletteName: palette.name,
          color: palette.color_1,
          dbKey: 'color_1'
        },
        {
          paletteID: palette.id,
          paletteName: palette.name,
          color: palette.color_2,
          dbKey: 'color_2'
        },
        {
          paletteID: palette.id,
          paletteName: palette.name,
          color: palette.color_3,
          dbKey: 'color_3'
        },
        {
          paletteID: palette.id,
          paletteName: palette.name,
          color: palette.color_4,
          dbKey: 'color_4'
        },
        {
          paletteID: palette.id,
          paletteName: palette.name,
          color: palette.color_5,
          dbKey: 'color_5'
        }
      ]
      return (
        <>
        <h2 
          contentEditable={true} 
          onBlur={(e) => this.patchProject(`palettes/${palette.id}`, { name: e.target.innerText })}
        >
        {palette.name}
        </h2>
          <div onClick={()=> this.deletePalette(palette.id)}>
            ðŸ—‘
          </div>

          <div style={{display:'flex', flexWrap:'wrap', width:'100%'}}>
            {colors.map(colorObj => {
              return <ColorPreview 
                color={colorObj.color} 
                paletteID={colorObj.paletteID} 
                dbKey={colorObj.dbKey}
                patchProject={this.patchProject}
              />
            })
            }
          </div>
        </>
      )}
    )
  }

  render() {
    return (
      <div className='palettes' style={{overflow: "auto"}}>
        <Link to='/projects'>
          <span role="img" aria-label="back buttton">ðŸ”™</span>
        </Link>
        <h2>
          {this.state.project_name}
        </h2>
        {this.showPalettes()}
      </div>
    )
  }
}

export default SingleProject
