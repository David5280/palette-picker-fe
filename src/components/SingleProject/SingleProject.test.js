import React from 'react';
import { SingleProject } from './SingleProject';
import { shallow } from 'enzyme';

jest.mock('../../fetchCalls/fetchCalls', () => ({
  serverCall: jest.fn().mockImplementation(() => {
    return [{ id: 1, name: 'title1',  }, { id: 2, title: 'title2' }];
  })
}));

describe('SingleProject', () => {
  let wrapper;
  let mockProjectID;
  beforeEach(() => {
    mockProjectID = 3
    wrapper = shallow(<SingleProject id={mockProjectID}/>)
    
  })
  it('should do match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});