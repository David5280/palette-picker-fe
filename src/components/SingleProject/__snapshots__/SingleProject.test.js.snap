// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleProject should do match the snapshot 1`] = `ShallowWrapper {}`;
